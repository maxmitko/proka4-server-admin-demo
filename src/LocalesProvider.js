import { locales } from 'locales'
import React from 'react';

export const LocalesContext = React.createContext();

export default props => <LocalesContext.Provider value={locales} children={props.children} />