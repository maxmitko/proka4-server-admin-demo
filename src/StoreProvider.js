import React from 'react';
import { applyMiddleware, compose, createStore } from 'redux'
import { StoreContext } from 'redux-react-hook';
import createSagaMiddleware from 'redux-saga'
import thunk from 'redux-thunk';
import reducers from 'store/reducers';
import rootSaga from 'store/sagas'

const sagaMiddleware = createSagaMiddleware();
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
    reducers,
    composeEnhancers(applyMiddleware(thunk, sagaMiddleware)),
);

sagaMiddleware.run(rootSaga);

const StoreProvider = props => <StoreContext.Provider value={store} children={props.children} />

export default StoreProvider