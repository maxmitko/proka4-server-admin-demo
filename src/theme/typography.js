
const defaultFontFamily = "'Roboto', sans-serif"

export default {
    h1: {
        fontWeight: 300,
        fontSize: 6,
        letterSpacing: -1.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h2: {
        fontWeight: 300,
        fontSize: 3.75,
        letterSpacing: -0.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h3: {
        fontWeight: 400,
        fontSize: 3,
        letterSpacing: 0,
        lineHeight: 1.04,
        fontFamily: defaultFontFamily,
        tag: "h3",
    },
    h4: {
        fontWeight: 400,
        fontSize: 2.125,
        letterSpacing: 0.25,
        lineHeight: 1.17,
        fontFamily: defaultFontFamily,
        tag: "h4",
    },
    h5: {
        fontWeight: 400,
        fontSize: 1.5,
        letterSpacing: 0,
        lineHeight: 1.33,
        fontFamily: defaultFontFamily,
        tag: "h5",
    },
    h6: {
        fontWeight: 300,
        fontSize: 1.15,
        letterSpacing: 0,
        lineHeight: 1.4,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.15,
        lineHeight: 1.75,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle2: {
        fontWeight: 500,
        fontSize: 0.875,
        letterSpacing: 0.1,
        lineHeight: 1.57,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    body1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.5,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    body2: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 0.25,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    button: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 1.25,
        lineHeight: 1.5,
        fontFamily: "'Roboto Condensed', sans-serif",
        tag: "span",
    },
    caption: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 0.4,
        lineHeight: 1.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    overline: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 1.5,
        lineHeight: 2.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    inherit: {
        fontWeight: "inherit",
        fontSize: "inherit",
        letterSpacing: "inherit",
        lineHeight: "inherit",
        fontFamily: "inherit",
        tag: "span",
    },
}