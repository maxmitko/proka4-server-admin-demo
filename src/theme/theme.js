import typography from './typography'
import produce from "immer"

export const theme = {
    palette: {
        primary: "hsla(0, 0%, 0%, 1)",
        secondary: "hsla(187, 100%, 42%, 1)",
        background: "hsla(0, 0%, 93%, 1)",
        surface: "hsla(0, 0%, 100%, 1)",
        error: "hsla(0, 100%, 42%, 1)",
    },
    paletteOn: {
        primary: "hsla(0, 0%, 100%, 1)",
        secondary: "hsla(0, 0%, 100%, 1)",
        background: "hsla(0, 0%, 0%, 1)",
        surface: "hsla(210, 9%, 31%, 1)",
        error: "hsla(0, 0%, 100%, 1)",
    },
    extra: {
        default: null,
        gray: "hsla(0, 0%, 76%, 1)",
        primary: "hsla(291, 64%, 42%, 1)",
        danger: "hsla(4, 90%, 58%, 1)",
        success: "hsla(123, 36%, 52%, 1)",
        warning: "hsla(34, 98%, 51%, 1)",
        info: "hsla(187, 100%, 42%, 1)",
        rose: "hsla(340, 82%, 58%, 1)",
        white: "hsla(0, 0%, 100%, 1)",
    },
    extraOn: {
        default: null,
        gray: "hsla(0, 0%, 100%, 1)",
        primary: "hsla(0, 0%, 100%, 1)",
        danger: "hsla(0, 0%, 100%, 1)",
        success: "hsla(0, 0%, 100%, 1)",
        warning: "hsla(0, 0%, 100%, 1)",
        info: "hsla(0, 0%, 100%, 1)",
        rose: "hsla(0, 0%, 100%, 1)",
        white: "hsla(0, 0%, 0%, 1)",
    },
    extraLight: {
        default: null,
        gray: "hsla(0, 0%, 72%, 1)",
        primary: "hsla(291, 88%, 57%, 1)",
        danger: "hsla(22, 100%, 53%, 1)",
        success: "hsla(123, 45%, 57%, 1)",
        warning: "hsla(34, 98%, 60%, 1)",
        info: "hsla(187, 95%, 46%, 1)",
        rose: "hsla(340, 82%, 65%, 1)",
        white: "hsl(0, 100%, 100%)",
    },
    extraShadow: {
        default: null,
        gray: "hsla(0, 0%, 30%, 0.22)",
        primary: "hsla(291, 64%, 54%, 0.25)",
        danger: "hsla(0, 100%, 58%, 0.25)",
        success: "hsla(123, 45%, 57%, 0.25)",
        warning: "hsla(34, 98%, 60%, 0.25)",
        info: "hsla(187, 95%, 46%, 0.25)",
        rose: "hsla(340, 82%, 63%, 0.25)",
        white: "hsla(0, 0%, 30%, 0.25)",
    },
    typography: typography,
    zIndex: {
        mobileStepper: 1000,
        appBar: 1100,
        drawer: 1200,
        modal: 1300,
        snackbar: 1400,
        tooltip: 1500,
    }
}

const newTheme = produce(theme, draftTheme => {
    draftTheme.extra.default = theme.extra.primary
    draftTheme.extraOn.default = theme.extraOn.primary
    draftTheme.extraLight.default = theme.extraLight.primary
    draftTheme.extraShadow.default = theme.extraShadow.primary
})


export default newTheme