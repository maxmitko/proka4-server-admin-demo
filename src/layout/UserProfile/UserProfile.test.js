import LocalesProvider from 'LocalesProvider'
import StoreProvider from 'StoreProvider'
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components'
import theme from 'theme/theme'

import userProfileMock from '../../__mocks__/userProfileMock'
import TestComponent from './UserProfile';

const withHOC = ui => (
    <StoreProvider>
        <LocalesProvider>
            <ThemeProvider theme={theme}>
                {ui}
            </ThemeProvider >
        </LocalesProvider>
    </StoreProvider>
)


describe('enzyme UserProfile', () => {

    let wrapper;
    let element;
    const mockData = userProfileMock.initialData

    beforeEach(() => {
        element = withHOC(<TestComponent {...userProfileMock} />)
    });

    afterEach(() => {
        wrapper && wrapper.unmount();
    });

    it('default SNAPSHOT should render correctly', () => {
        const tree = renderer.create(element).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('check fields values', () => {
        wrapper = mount(element);

        const formFields = ["username", "fullname", "email", "money", "phone"]

        formFields.forEach(item => {
            expect(wrapper.find({ value: mockData[item] })).toBeDefined()
        })
    });
});