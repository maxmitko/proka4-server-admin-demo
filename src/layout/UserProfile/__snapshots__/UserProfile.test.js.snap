// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme UserProfile default SNAPSHOT should render correctly 1`] = `
.c9 {
  border-radius: 0.2rem;
  cursor: pointer;
  min-height: auto;
  text-transform: uppercase;
  border: none;
  outline: none;
  color: hsla(0,0%,100%,1);
  font-family: 'Roboto Condensed',sans-serif;
  font-size: 0.875rem;
  -webkit-transition: 200ms cubic-bezier(0.4,0,0.2,1);
  transition: 200ms cubic-bezier(0.4,0,0.2,1);
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.875rem;
  padding: 12px 17px;
  background-color: hsla(123,36%,52%,1);
  box-shadow: 0 1px 3px 0px hsla(0,0%,30%,0.4);
}

.c9:disabled,
.c9:hover:disabled {
  opacity: 0.75;
  cursor: auto;
  box-shadow: none;
}

.c9:hover {
  background-color: hsla(123,45%,57%,1);
  box-shadow: 0 3px 4px 0px hsla(123,45%,57%,0.25), 2px 0px 4px 0px hsla(123,45%,57%,0.25), -2px 0px 4px 0px hsla(123,45%,57%,0.25);
}

.c9:disabled,
.c9:hover:disabled {
  background-color: hsla(123,36%,52%,1);
}

.c11 {
  position: absolute !important;
  border-radius: inherit;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c10 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  position: relative;
  margin: 0;
  padding: 0;
  border: 0;
  min-width: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Roboto',sans-serif;
  font-size: 0.875rem;
  width: 100%;
  height: 32px;
  color: hsla(210,9%,31%,1);
  border-bottom: 1px solid;
  border-color: hsl(0,0%,67%);
  height: 28px;
}

.c6 {
  border-bottom: 1px solid hsl(0,0%,67%);
}

.c6::after {
  content: "";
  left: 0;
  right: 0;
  bottom: -1px;
  position: absolute;
  pointer-events: none;
  border-bottom: 2px solid hsla(291,64%,42%,1);
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition: 200ms;
  transition: 200ms;
  pointer-events: none;
}

.c3 {
  font-weight: 300;
  color: hsl(0,0%,67%);
  position: relative;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  cursor: inherit;
}

.c2 label {
  margin-right: 0.8rem;
}

.c5 {
  cursor: inherit;
  text-align: inherit;
  font-size: inherit;
  position: relative;
  background-color: transparent;
  width: 100%;
  outline: none;
  border: none;
  color: hsla(210,9%,31%,1);
}

.c5::-webkit-input-placeholder {
  color: hsla(0,0%,67%,0.7);
  opacity: 1;
}

.c5::-moz-placeholder {
  color: hsla(0,0%,67%,0.7);
  opacity: 1;
}

.c5:-ms-input-placeholder {
  color: hsla(0,0%,67%,0.7);
  opacity: 1;
}

.c5::placeholder {
  color: hsla(0,0%,67%,0.7);
  opacity: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 20px 0;
}

.c1 label {
  min-width: 140px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-right: 30px;
  text-align: right;
}

.c1 > div {
  margin: 15px 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 20px;
}

.c7 > * {
  margin-left: 20px;
}

.c8 svg {
  fill: white;
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      color="default"
      label="Логин"
      onChange={[Function]}
      value="sweet"
    >
      <label
        className="c3"
      >
        Логин
      </label>
      <div
        className="c4"
        color="default"
        onClick={[Function]}
      >
        <input
          className="c5"
          onBlur={[Function]}
          onChange={[Function]}
          value="sweet"
        />
        <div
          className="c6"
          color="default"
        />
      </div>
    </div>
    <div
      className="c2"
      color="default"
      label="ФИО"
      onChange={[Function]}
      value="Моисеева Ульяна Дмитриевна"
    >
      <label
        className="c3"
      >
        ФИО
      </label>
      <div
        className="c4"
        color="default"
        onClick={[Function]}
      >
        <input
          className="c5"
          onBlur={[Function]}
          onChange={[Function]}
          value="Моисеева Ульяна Дмитриевна"
        />
        <div
          className="c6"
          color="default"
        />
      </div>
    </div>
    <div
      className="c2"
      color="default"
      label="Почта"
      onChange={[Function]}
      value="ovosh@mail.ru"
    >
      <label
        className="c3"
      >
        Почта
      </label>
      <div
        className="c4"
        color="default"
        onClick={[Function]}
      >
        <input
          className="c5"
          onBlur={[Function]}
          onChange={[Function]}
          value="ovosh@mail.ru"
        />
        <div
          className="c6"
          color="default"
        />
      </div>
    </div>
    <div
      className="c2"
      color="default"
      label="Кошелек (руб)"
      onChange={[Function]}
      value={1200}
    >
      <label
        className="c3"
      >
        Кошелек (руб)
      </label>
      <div
        className="c4"
        color="default"
        onClick={[Function]}
      >
        <input
          className="c5"
          onBlur={[Function]}
          onChange={[Function]}
          value={1200}
        />
        <div
          className="c6"
          color="default"
        />
      </div>
    </div>
    <div
      className="c2"
      color="default"
      label="Телефон"
      onChange={[Function]}
      value="+79829998733"
    >
      <label
        className="c3"
      >
        Телефон
      </label>
      <div
        className="c4"
        color="default"
        onClick={[Function]}
      >
        <input
          className="c5"
          onBlur={[Function]}
          onChange={[Function]}
          value="+79829998733"
        />
        <div
          className="c6"
          color="default"
        />
      </div>
    </div>
    <div
      className="c2"
      color="default"
      label="Пароль"
      onChange={[Function]}
      type="password"
      value=""
    >
      <label
        className="c3"
      >
        Пароль
      </label>
      <div
        className="c4"
        color="default"
        onClick={[Function]}
      >
        <input
          className="c5"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
        <div
          className="c6"
          color="default"
        />
      </div>
    </div>
    <div
      className="c2"
      color="default"
      label="Повторите пароль"
      onChange={[Function]}
      type="password"
      value=""
    >
      <label
        className="c3"
      >
        Повторите пароль
      </label>
      <div
        className="c4"
        color="default"
        onClick={[Function]}
      >
        <input
          className="c5"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
        <div
          className="c6"
          color="default"
        />
      </div>
    </div>
  </div>
  <div
    className="c7"
  >
    <button
      className="c8 c9"
      color="success"
      size="medium"
      type="submit"
    >
      <div
        className="c10"
      >
        Сохранить
      </div>
      <div
        className="rtr-root c11"
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        timeout={
          Object {
            "enter": 500,
            "exit": 500,
          }
        }
      >
        <span />
      </div>
    </button>
  </div>
</form>
`;
